catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswStore.java

```diff
@@ -0,0 +1,344 @@
+import net.opengis.cat.csw.v_2_0_2.InsertResultType;
+import net.opengis.cat.csw.v_2_0_2.QueryConstraintType;
+import net.opengis.cat.csw.v_2_0_2.TransactionResponseType;
+import net.opengis.cat.csw.v_2_0_2.dc.elements.SimpleLiteral;
+import net.opengis.filter.v_1_1_0.FilterType;
+
+public abstract class AbstractCswStore extends AbstractCswSource implements CatalogStore {
+
+    protected TransformerManager schemaTransformerManager;
+
+    /**
+     * Instantiates a CswStore. This constructor is for unit tests
+     *
+     * @param context                The {@link BundleContext} from the OSGi Framework
+     * @param cswSourceConfiguration the configuration of this source
+     * @param provider               transform provider to transform results
+     * @param factory                client factory already configured for this source
+     */
+    public AbstractCswStore(BundleContext context, CswSourceConfiguration cswSourceConfiguration,
+            Converter provider, SecureCxfClientFactory factory) {
+        super(context, cswSourceConfiguration, provider, factory);
+    }
+
+    /**
+     * Instantiates a CswStore.
+     */
+    public AbstractCswStore() {
+        super();
+    }
+
+    @Override
+    public CreateResponse create(CreateRequest createRequest) throws IngestException {
+        Map<String, Serializable> properties = new HashMap<>();
+
+        validateOperation();
+
+        Subject subject =
+                (Subject) createRequest.getPropertyValue(SecurityConstants.SECURITY_SUBJECT);
+        Csw csw = factory.getClientForSubject(subject);
+        CswTransactionRequest transactionRequest = getTransactionRequest();
+
+        List<Metacard> metacards = createRequest.getMetacards();
+        List<String> metacardIds = metacards.stream()
+                .map(mcard -> mcard.getId())
+                .collect(Collectors.toList());
+        ArrayList<Metacard> createdMetacards = new ArrayList<>();
+        ArrayList<Filter> createdMetacardFilters = new ArrayList<>();
+        HashSet<ProcessingDetails> errors = new HashSet<>();
+
+        String insertTypeName = schemaTransformerManager.getTransformerIdForSchema(
+                cswSourceConfiguration.getOutputSchema());
+
+        if (insertTypeName == null) {
+            throw new IngestException("Could not find transformer for output schema "
+                    + cswSourceConfiguration.getOutputSchema());
+        }
+
+        transactionRequest.getInsertActions()
+                .add(new InsertAction(insertTypeName, null, metacards));
+        try {
+            TransactionResponseType response = csw.transaction(transactionRequest);
+            Set<String> processedIds = new HashSet<>();
+            //dive down into the response to get the created ID's. We need these so we can query
+            //the source again to get the created metacards and put them in the result
+            createdMetacardFilters.addAll(response.getInsertResult()
```

Comment:
You could just: `createdMetacardFilters = response.getInsertResult().stream()...;` and skip the extra loop to add them all.